route/route.go

```diff
@@ -22,9 +22,13 @@ import (
 
 type param string
 
-// Param returns param p for the context.
+// Param returns param p for the context, or the empty string when
+// param does not exist in context.
 func Param(ctx context.Context, p string) string {
-	return ctx.Value(param(p)).(string)
+	if v, ok := ctx.Value(param(p)).(string); ok {
```

Comment:
This would mask if somebody put a non-string as value. How about this:
```Go
func Param(ctx context.Context, p string) string {
	if v := ctx.Value(param(p)); v!=nil {
		return v.(string)
	}
￼ 	return ""
}
```

Reply:
Non-string as value is not possible outside of the package since `WithParam` accepts a `p, v string`, and the type `param` is package private.

Reply:
You could still put something else into the context by directly accessing it.

But even without that, I think "xxx is impossible" is a good reason to panic if xxx happens anyway.

Reply:
So both, my and your argument are arguments to apply my suggestion. :o)

Reply:
> You could still put something else into the context by directly accessing it.

Outside of this package, because `type param string` creates a unique, package private type, you can't inject something into the Context that can be _read_ by `Param`. Other packages that have access to the context passed in, it's true, *can* add values to the context, but they can't do so _and_ be visible to `Param`.